drivers/gles2/rasterizer_storage_gles2.cpp

```diff
@@ -5156,14 +5156,22 @@ void RasterizerStorageGLES2::render_target_set_external_texture(RID p_render_tar
 
 		// is there a point to setting the internal formats? we don't know them..
 
-		// set our texture as the destination for our framebuffer
-		glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, p_texture_id, 0);
-
-		// seeing we're rendering into this directly, better also use our depth buffer, just use our existing one :)
-		if (config.support_depth_texture) {
-			glFramebufferTexture2D(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_TEXTURE_2D, rt->depth, 0);
+		// check if MSAA is active to set the correct depth buffer and target texture
+		if (rt->multisample_active) {
+			static const int msaa_value[] = { 0, 2, 4, 8, 16 };
+			int msaa = msaa_value[rt->msaa];
+			glFramebufferRenderbuffer(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_RENDERBUFFER, rt->multisample_depth);
+			glFramebufferTexture2DMultisample(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, p_texture_id, 0, msaa);
```

Comment:
This only compiles on Android, other platforms don't fetch the extension for `glFramebufferTexture2DMultisample`.
Looking at other uses of this function, only Android seems to need using a texture for MSAA, other platforms use a renderbuffer directly.

Reply:
ok; so I can just check for `ANDROID_ENABLED` in this case?